default.nix

```diff
@@ -10,44 +10,35 @@ let
       imports = lib.singleton { disko.devices = cfg.disko.devices; };
       options = {
         disko.devices = lib.mkOption {
-          type = diskoLib.devices;
+          type = diskoLib.toplevel;
         };
       };
     };
   };
 in
 {
   lib = diskoLib;
-  create = cfg: diskoLib.create (eval cfg).config.disko.devices;
-  createScript = cfg: pkgs: (diskoLib.writeCheckedBash { inherit pkgs checked; }) "disko-create" ''
-    export PATH=${lib.makeBinPath (diskoLib.packages (eval cfg).config.disko.devices pkgs)}:$PATH
-    ${diskoLib.create (eval cfg).config.disko.devices}
-  '';
-  createScriptNoDeps = cfg: pkgs: (diskoLib.writeCheckedBash { inherit pkgs checked; noDeps = true; }) "disko-create" ''
-    ${diskoLib.create (eval cfg).config.disko.devices}
-  '';
-  mount = cfg: diskoLib.mount (eval cfg).config.disko.devices;
-  mountScript = cfg: pkgs: (diskoLib.writeCheckedBash { inherit pkgs checked; }) "disko-mount" ''
-    export PATH=${lib.makeBinPath (diskoLib.packages (eval cfg).config.disko.devices pkgs)}:$PATH
-    ${diskoLib.mount (eval cfg).config.disko.devices}
-  '';
-  mountScriptNoDeps = cfg: pkgs: (diskoLib.writeCheckedBash { inherit pkgs checked; noDeps = true; }) "disko-mount" ''
-    ${diskoLib.mount (eval cfg).config.disko.devices}
-  '';
-  disko = cfg: diskoLib.zapCreateMount (eval cfg).config.disko.devices;
-  diskoScript = cfg: pkgs: (diskoLib.writeCheckedBash { inherit pkgs checked; }) "disko-zap-create-mount" ''
-    export PATH=${lib.makeBinPath ((diskoLib.packages (eval cfg).config.disko.devices pkgs) ++ [ pkgs.bash ])}:$PATH
-    ${diskoLib.zapCreateMount (eval cfg).config.disko.devices}
-  '';
+
+  # legacy alias
+  create = cfg: builtins.trace "the create output is deprecated, use format instead" (eval cfg).config.disko.devices._create;
+  createScript = cfg: pkgs: builtins.trace "the create output is deprecated, use format instead" ((eval cfg).config.disko.devices._scripts { inherit pkgs checked; }).formatScript;
+  createScriptNoDeps = cfg: pkgs: builtins.trace "the create output is deprecated, use format instead" ((eval cfg).config.disko.devices._scripts { inherit pkgs checked; }).formatScriptNoDeps;
+
+  format = cfg: (eval cfg).config.disko.devices._create;
+  formatScript = cfg: pkgs: ((eval cfg).config.disko.devices._scripts { inherit pkgs checked; }).formatScript;
+  formatScriptNoDeps = cfg: pkgs: ((eval cfg).config.disko.devices._scripts { inherit pkgs checked; }).formatScriptNoDeps;
+
+  mount = cfg: (eval cfg).config.disko.devices._mount;
+  mountScript = cfg: pkgs: ((eval cfg).config.disko.devices._scripts { inherit pkgs checked; }).mountScript;
+  mountScriptNoDeps = cfg: pkgs: ((eval cfg).config.disko.devices._scripts { inherit pkgs checked; }).mountScriptNoDeps;
+
+  disko = cfg: (eval cfg).config.disko.devices._disko;
```

Comment:
so disko is create+format+mount?

Reply:
create is an alias for format. disko is umount + format + mount

Reply:
As I just learned it's umount + destroy + format + mount